src/configure/clients/github/githubClient.ts

```diff
@@ -36,41 +36,38 @@ export class GithubClient {
         await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
     }
 
-    public async createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+    public async createGithubRepo(orgName: string, repoName: string): Promise<GitHubRepo> {
         let restClient = new RestClient();
         let Url = "https://api.github.com/orgs/" + orgName + "/repos";
-        try {
-            return restClient.sendRequest(<UrlBasedRequestPrepareOptions>{
-                url: Url,
-                headers: {
-                    "Content-Type": "application/json",
-                    "Authorization": "Bearer " + this.patToken,
-                    "User-Agent": UserAgent
-                },
-                method: 'POST',
-                body: {
-                    "name": repoName,
-                    "description": "Repo created from VScode extension 'Deploy to Azure'",
-                    "homepage": "https://github.com",
-                    "private": false,
-                    "has_issues": true,
-                    "has_projects": true,
-                    "has_wiki": true
-                },
-                deserializationMapper: null,
-                serializationMapper: null
-            })
-                .then((detail: GitHubRepo) => {
-                    return detail;
-                }).catch(error => {
-                    telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
+        return restClient.sendRequest3(<UrlBasedRequestPrepareOptions>{
+            url: Url,
+            headers: {
+                "Content-Type": "application/json",
+                "Authorization": "Bearer " + this.patToken,
+                "User-Agent": UserAgent
+            },
+            method: 'POST',
+            body: {
+                "name": repoName,
+                "description": "Repo created from VScode extension 'Deploy to Azure'",
+                "homepage": "https://github.com",
+                "private": true,
+                "has_issues": true,
+                "has_projects": true,
+                "has_wiki": true
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+            .then((detail: GitHubRepo) => {
+                return detail;
+            }).catch(error => {
+                telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
+                if (error.response.statusCode === 422) {
                     return null;
-                });
-        }
-        catch (error) {
-            telemetryHelper.logError(Layer, TracePoints.GitHubRepositoryCreationFailed, error);
-            return null;
-        }
+                }
```

Comment:
should you throw an error rather than returning null. This would ensure consistency in the function's behavior in case something goes wrong. Right now it is throowing error in some case and returning null in some.

Reply:
`null` is returned only in the case where API call needs to be made again (same repo name exists). If the error is directly thrown then in the calling function there will be no error code to check the reason for failure.